data_specifications/specification.proto

```diff
@@ -0,0 +1,195 @@
+syntax = "proto3";
+
+package specification;
+
+enum Arch {
+  ARCH_UNSPECIFIED = 0;
+  ARCH_X86 = 1;
+  ARCH_X86_AVX = 2;
+  ARCH_X86_AVX512 = 3;
+  ARCH_AMD64 = 4;
+  ARCH_AMD64_AVX = 5;
+  ARCH_ADM64_AVX512 = 6;
+  ARCH_AARCH64 = 7;
+  ARCH_AARCH32 = 8;
+  ARCH_SPARC32 = 9;
+  ARCH_SPARC64 = 10;
+}
+
+enum OS {
+  OS_UNSPECIFIED = 0;
+  OS_LINUX = 1;
+  OS_MACOS = 2;
+  OS_WINDOWS = 3;
+  OS_SOLARIS = 4;
+}
+
+enum BaseType {
+  BT_BOOL = 0;
+  BT_CHAR = 1;
+  BT_SCHAR = 2;
+  BT_UCHAR = 3;
+  BT_I8 = 4;
+  BT_U8 = 5;
+  BT_I16 = 6;
+  BT_U16 = 7;
+  BT_I24 = 8;
+  BT_U24 = 9;
+  BT_I32 = 10;
+  BT_U32 = 11;
+  BT_I64 = 12;
+  BT_U64 = 13;
+  BT_I128 = 14;
+  BT_U128 = 15;
+  BT_FL16 = 16;
+  BT_FL32 = 17;
+  BT_FL64 = 18;
+  BT_FL80 = 19;
+  BT_FL96 = 20;
+  BT_FL128 = 21;
+  BT_M64 = 22;
+  BT_VOID = 23;
+  BT_PADDING = 24;
+}
+
+message TypeSpec {
+  message PointerType {
+    TypeSpec pointee = 1;
```

Comment:
When we translate these to LLVM types, we're going to lose the pointee type. Do we need this?

Reply:
It's useful for when we use the spec to guide Rellic